File: py-sdk/mylogger/services/alert.py
from ..core.record import LogRecord, Level

class AlertService:
    def evaluate(self, record: LogRecord):
        if record.level >= Level.FATAL:
            self._trigger_pagerduty(record)
        
        # Domain specific rule logic
        if record.context.get("deadlock") is True:
            self._trigger_slack(record, "#db-alerts")

    def _trigger_pagerduty(self, record: LogRecord):
        print(f"[ALERT] PagerDuty firing for {record.service}: {record.message}")

    def _trigger_slack(self, record: LogRecord, channel: str):
        print(f"[ALERT] Slack message to {channel}: {record.message}")

File: ts-sdk/src/services/alert.ts
import { LogRecord } from '../core/record';
import { Level } from '../core/enums';

export class AlertService {
  evaluate(record: LogRecord): void {
    if (record.level >= Level.FATAL) {
      this.triggerPagerDuty(record);
    }

    if (record.context.deadlock === true) {
      this.triggerSlack(record, "#db-alerts");
    }
  }

  private triggerPagerDuty(record: LogRecord): void {
    console.error(`[ALERT] PagerDuty firing for ${record.service}: ${record.message}`);
  }

  private triggerSlack(record: LogRecord, channel: string): void {
    console.error(`[ALERT] Slack message to ${channel}: ${record.message}`);
  }
}
